cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/class-media.php

```diff
@@ -612,13 +611,17 @@ public function get_transformations_from_string( $str, $type = 'image' ) {
 				$item = trim( $item );
 				foreach ( $params as $param => $type ) {
 					if ( substr( $item, 0, strlen( $param ) + 1 ) === $param . '_' ) {
-						$transformations[ $index ][ $type ] = substr( $item, strlen( $param ) + 1 );
+						if ( false === $return_string ) {
+							$transformations[ $index ][ $type ] = substr( $item, strlen( $param ) + 1 );
+						} else {
+							$transformations[] = $item;
+						}
 					}
 				}
 			}
 		}
 
-		return array_values( $transformations ); // Reset the keys.
+		return $return_string ? implode( ',', $transformations ) : array_values( $transformations );
```

Comment:
@dugajean as mentioned in a previous comment, there is a function in `API:: generate_transformation_string` that can handle this.
Slashes have purpose so we need to keep them to separate parts. This function does it correctly. Perhaps use that function here.